backend/src/Controllers/PostsImportController.php

```diff
@@ -0,0 +1,337 @@
+						/* translators: %s: post/page title */
+						'post_title'     => $property['TITLE'],
+						'post_type'      => $property['WP:POST_TYPE'],
+						'to_ping'        => $property['WP:PING_STATUS'],
+						'menu_order'     => $property['WP:MENU_ORDER'],
+					];
+
+					if ( ! function_exists( 'post_exists' ) ) {
+						require_once ABSPATH . 'wp-admin/includes/post.php';
+					}
+					if ( 0 === post_exists( $property['TITLE'], '', '', $property['WP:POST_TYPE'] ) ) {
+
+						/* Import Attachments */
+
+						if ( $property['WP:POST_TYPE'] == 'attachment' ) {
+
+							$file = $property['WP:ATTACHMENT_URL'];
+							$filename = basename( $file );
+							$read_file = file_get_contents( $file );
+
+							if ( $read_file ) {
+
+								$upload_file = wp_upload_bits( $filename, null, $read_file );
+
+								if ( ! $upload_file['error'] ) {
+
+									$wp_filetype = wp_check_filetype( $filename, null );
+									$post_data['post_mime_type'] = $wp_filetype['type'];
+									$attachment_id = wp_insert_attachment( $post_data, $upload_file['file'] );
+									if ( ! is_wp_error( $attachment_id ) ) {
+										require_once( ABSPATH . 'wp-admin' . '/includes/image.php' );
+										$attachment_data = wp_generate_attachment_metadata( $attachment_id, $upload_file['file'] );
+										wp_update_attachment_metadata( $attachment_id, $attachment_data );
+									}
+
+									$sucess_imp_count++;
+
+								} else {
+									$fail_imp_count++;
+									return rest_ensure_response(
+										[
+											'error'   => true,
+											'message' => 'Unable to upload attachment!',
+										]
+									);
+								}
+							} else {
+								$fail_imp_count++;
+								return rest_ensure_response(
+									[
+										'error'   => true,
+										'message' => 'Unable to upload attachment!',
+									]
+								);
+							}
+						} else {
+							$new_post_id = wp_insert_post( $post_data );
+							wp_set_post_terms( $new_post_id, null, 'category' );
+							$post_meta = $property['WP:POSTMETA'];
+							$post_cat  = $property['CATEGORY'];
+
+							if ( count( $post_meta ) !== 0 ) {
+								foreach ( $post_meta as $meta_info ) {
+
+									$meta_key   = $meta_info['WP:META_KEY'];
+									$meta_value = addslashes( $meta_info['WP:META_VALUE'] );
+									$wpdb->query( "INSERT INTO {$wpdb->postmeta} (post_id, meta_key, meta_value) values ({$new_post_id}, '{$meta_key}', '{$meta_value}')" ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+
+								}
+							}
+
+							if ( count( $post_cat ) !== 0 ) {
+
+								foreach ( $post_cat as $category ) {
+
+									   $term = term_exists( $category['content'], $category['DOMAIN'] );
+
+									   wp_set_post_terms( $new_post_id, [ $term['term_taxonomy_id'] ], $category['DOMAIN'], true );
+
+								}
+							}
+
+							$sucess_imp_count++;
+						}
+					} else {
+						$fail_imp_count++;
+
+					}
+				}
+
+				$post_fail_msg = 'Post Imported Failed!';
```

Comment:
In general, let's only fail if the post can't be imported. If attachments or terms fail, let's still let the import complete.

Reply:
Done